server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -143,291 +126,48 @@ public LookupCoordinatorManager(
-    }
-    final List<ListenableFuture<?>> futures = new ArrayList<>(urls.size());
-    for (final URL url : urls) {
-      futures.add(executorService.submit(new Runnable()
-      {
-        @Override
-        public void run()
-        {
-          try {
-            updateAllOnHost(url, knownLookups);
-          }
-          catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            LOG.warn("Update on [%s] interrupted", url);
-            throw Throwables.propagate(e);
-          }
-          catch (IOException | ExecutionException e) {
-            // Don't raise as ExecutionException. Just log and continue
-            LOG.makeAlert(e, "Error submitting to [%s]", url).emit();
-          }
-        }
-      }));
-    }
-    final ListenableFuture allFuture = Futures.allAsList(futures);
-    try {
-      allFuture.get(lookupCoordinatorManagerConfig.getUpdateAllTimeout().getMillis(), TimeUnit.MILLISECONDS);
-    }
-    catch (TimeoutException e) {
-      LOG.warn("Timeout in updating hosts! Attempting to cancel");
-      // This should cause Interrupted exceptions on the offending ones
-      allFuture.cancel(true);
-    }
-  }
-
-  Collection<URL> getAllHostsAnnounceEndpoint(final String tier) throws IOException
-  {
-    return ImmutableList.copyOf(
-        Collections2.filter(
-            Collections2.transform(
-                listenerDiscoverer.getNodes(LookupModule.getTierListenerPath(tier)),
-                HOST_TO_URL
-            ),
-            Predicates.notNull()
-        )
-    );
   }
 
   public boolean updateLookup(
       final String tier,
       final String lookupName,
-      Map<String, Object> spec,
+      LookupExtractorFactoryMapContainer spec,
       final AuditInfo auditInfo
   )
   {
     return updateLookups(
-        ImmutableMap.<String, Map<String, Map<String, Object>>>of(tier, ImmutableMap.of(lookupName, spec)),
+        ImmutableMap.<String, Map<String, LookupExtractorFactoryMapContainer>>of(tier, ImmutableMap.of(lookupName, spec)),
         auditInfo
     );
   }
 
-  public boolean updateLookups(final Map<String, Map<String, Map<String, Object>>> updateSpec, AuditInfo auditInfo)
+  public boolean updateLookups(final Map<String, Map<String, LookupExtractorFactoryMapContainer>> updateSpec, AuditInfo auditInfo)
   {
-    synchronized (startStopSync) {
-      final Map<String, Map<String, Map<String, Object>>> priorSpec = getKnownLookups();
+    Preconditions.checkState(lifecycleLock.awaitStarted(5, TimeUnit.SECONDS), "not started");
+
+    if (updateSpec.isEmpty() && lookupMapConfigRef.get() != null) {
+      return true;
+    }
+
+    //ensure all the lookups specs have version specified. ideally this should be done in the LookupExtractorFactoryMapContainer
+    //constructor but that allows null to enable backward compatibility with 0.10.0 lookup specs
+    for (final String tier : updateSpec.keySet()) {
```

Comment:
@leventov ok, OTOH I don't think it matters really that much... these things could be debated for eternity but I sometimes address small review comments because they are very simple changes requiring much less time than arguing about it.